astropy/modeling/rotations.py

```diff
@@ -27,11 +27,93 @@
 from .parameters import Parameter
 
 
-__all__ = ['RotateCelestial2Native', 'RotateNative2Celestial', 'Rotation2D']
+__all__ = ['RotateCelestial2Native', 'RotateNative2Celestial', 'Rotation2D',
+           'EulerAngleRotation']
 
 
 class EulerAngleRotation(Model):
     """
+    Implements Euler angle rotation.
+
+    Parameters
+    ----------
+    phi, theta, psi : float
+        Euler angles in deg
+    order : str
+        A 3 character string, a combination of 'x', 'y' and 'z',
+        where each character denotes an axis in 3D space.
+    """
+
+    inputs = ('alpha', 'delta')
+    outputs = ('alpha', 'delta')
+
+    phi = Parameter(default=0, getter=np.rad2deg, setter=np.deg2rad)
+    theta = Parameter(default=0, getter=np.rad2deg, setter=np.deg2rad)
+    psi = Parameter(default=0, getter=np.rad2deg, setter=np.deg2rad)
+
+    def __init__(self, phi, theta, psi, order):
+        if len(order) >3 or len(order) < 2:
+            raise TypeError(
+                "Expected order to be a character sequence of 2 or 3, got {0}".format(order))
+        for i in order:
+            if i not in ['x', 'y', 'z']:
+                raise ValueError("Expected order to be a combination of characters"
+                                 "'x', 'y' and 'z', got {0}".format(order))
```

Comment:
Then if there were a `self.axes` this could be written:
`if set(order).difference(self.axes):` with the error string also updated accordingly.